src/cpp/session/modules/SessionWorkbench.cpp

```diff
@@ -68,8 +70,61 @@ namespace session {
 namespace modules { 
 namespace workbench {
 
-namespace {   
-      
+namespace {
+
+module_context::WaitForMethodFunction s_waitForActiveDocumentContext;
+
+SEXP rs_getActiveDocumentContext()
+{
+   // prepare context event
+   ClientEvent activeDocumentContextEvent(client_events::kGetActiveDocumentContext);
+   
+   // wait for event to complete
+   json::JsonRpcRequest request;
+   
+   bool succeeded = s_waitForActiveDocumentContext(&request, activeDocumentContextEvent);
+   if (!succeeded)
+      return R_NilValue;
+   
+   json::Object context = request.params[0].get_obj();
```

Comment:
If the client doesn't pass any params or if the first param isn't an object this will segfault. You could check the size and type of the first parameter, but `json::readObjectParam` will do this for you, e.g. https://github.com/rstudio/rstudio/blob/d5ab5dfa53c057a289179826a94622b715146e79/src/cpp/session/modules/tex/SessionSynctex.cpp#L236